### PR TITLE
chore: added olm servicemonitor manifests for the metrics services

### PR DESCRIPTION
**Pull request by TomerFi**
- Added _ServiceMonitor_ manifests for the _OLM Catalog_.
- Included _ServiceMonitor_ as a required _CRD_ for the operator _CSV_.
- ~~Updated metrics docs for clarification (removed unrelated misleading observability section).~~

> ~~_OLM_ manifests does not include the required _RBAC_ for collecting the metrics by _Promethues_'s _ServiceAccount_.~~

EDIT:
- Added _Role_ and _RoleBinding_ resources to allow _Promethues_ to scrap the new metric services.

issue: https://github.com/stolostron/backlog/issues/25649

### CODE REVIEW

File: docs/metrics.md
Comment (chenz4027) [inline docs/metrics.md:561]: Thanks for removing it, was confusing about this step

File: docs/metrics.md
Comment (chenz4027) [inline docs/metrics.md:226]: Is this changed?

**Review by chenz4027 — COMMENTED** [review]
/hold

@xiangjingli Could you help review? Thanks!

File: docs/metrics.md
Comment (TomerFi) [reply docs/metrics.md:226]: yes, besides the _ServiceMonitors_ which are included in this PR, all other resources already exist in _OCP_.
so I think, this doc will be used primarily by _K8S_ _OCM_ users, so this comment seems redundant to me.
I can bring it back if you think it's helpful.

Comment (TomerFi) [reply docs/metrics.md:226]: on second thought...
this will be useful for users who run _OCM_ on _OCP_.
so maybe we should indeed bring back...

your call.
:-)

Comment (chenz4027) [reply docs/metrics.md:226]: Yeah I think we should bring it back if there are other use cases using this note

Comment (TomerFi) [reply docs/metrics.md]: done.

**Comment by TomerFi** [comment]
@xiangjingli 
I brought back the observability section and added the required _Role_ and _RoleBinding_ for the metrics services scrapping.

In regards to observability in the upcoming playback, I'm not sure I can get my cluster ready with observability by tomorrow, we can split the playback and just do the observability part next week.

**Review by chenz4027 — APPROVED** [review]
/lgtm

**Pull request merged**
